src/libsyntax_ext/standard_library_imports.rs

```diff
@@ -55,32 +61,88 @@ pub fn inject(
     // the prelude.
     let name = names[0];
 
-    let span = DUMMY_SP.fresh_expansion(ExpnData::allow_unstable(
-        ExpnKind::Macro(MacroKind::Attr, sym::std_inject), DUMMY_SP, edition,
-        [sym::prelude_import][..].into(),
-    ));
+    let segments = if rust_2018 {
+        [name, sym::prelude, sym::v1].iter()
+            .map(|symbol| ast::PathSegment::from_ident(ast::Ident::new(*symbol, span)))
+            .collect()
+    } else {
+        [kw::PathRoot, name, sym::prelude, sym::v1].iter()
+            .map(|symbol| ast::PathSegment::from_ident(ast::Ident::new(*symbol, call_site)))
```

Comment:
Actually, `ast::Ident::new(*symbol, span)` should work here.
The reason is that `foo` in `::foo::bar` is [adjusted](https://github.com/rust-lang/rust/pull/62476/commits/0ec6ea7333a8918a96f40110e014f2bbbd72281f) to the root module before resolution.
Example: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2015&gist=976e386522a0bb4941dd541ec5dc5baf.